fem/integ/bilininteg_diffusion_kernels.hpp

```diff
@@ -152,98 +152,92 @@ inline void PADiffusionDiagonal2D(const int NE,
 }
 
 // Shared memory PA Diffusion Diagonal 2D kernel
-template<int T_D1D = 0, int T_Q1D = 0, int T_NBZ = 0>
-inline void SmemPADiffusionDiagonal2D(const int NE,
-                                      const bool symmetric,
-                                      const Array<double> &b_,
-                                      const Array<double> &g_,
-                                      const Vector &d_,
-                                      Vector &y_,
-                                      const int d1d = 0,
-                                      const int q1d = 0)
+template<int T_D1D, int T_Q1D, int T_NBZ>
+inline void SmemPADiffusionDiagonal2D_element(const int e,
```

Comment:
This and the other `*_element` functions are called from `MFEM_HOST_DEVICE` lambdas, so I suspect we need to make these `MFEM_HOST_DEVICE` as well.

Reply:
`done`